### ISSUE TITLE
Some warnings

### ISSUE THREAD

**Issue by dannywillems** [copilot-emacs/copilot.el#239]
I fixed some warnings, see PRs (https://github.com/copilot-emacs/copilot.el/pull/234, https://github.com/copilot-emacs/copilot.el/pull/235, https://github.com/copilot-emacs/copilot.el/pull/236, https://github.com/copilot-emacs/copilot.el/pull/237, https://github.com/copilot-emacs/copilot.el/pull/238), but there are still some:

```
In copilot--handle-notification:
copilot.el:469:14: Warning: ‘mark-whole-buffer’ is for interactive use only.

In end of data:
copilot.el:597:14: Warning: the function ‘vterm-insert’ is not known to be
    defined.
copilot.el:596:14: Warning: the function ‘vterm-delete-region’ is not known to
    be defined.
```

Emacs 29.1

**Comment by emil-vdw** [comment]
@dannywillems thanks for all the contributions! :tada:

**Comment by jcs090218** [comment]
All warnings have been addressed in #287.
